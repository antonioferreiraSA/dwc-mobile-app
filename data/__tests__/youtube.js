import axios from 'redaxios';
import { sortByDate, fetchChannelSection, fetchPlaylists } from '../youtube';

jest.mock('redaxios');

test('should sort playlist by publish date', () => {
  const items = [
    { id: '5', publishDate: '2020-04-20T00:00:00Z' },
    { id: '2', publishDate: '2020-04-05T00:00:00Z' },
    { id: '3', publishDate: '2020-04-10T00:00:00Z' },
    { id: '1', publishDate: '2020-04-01T00:00:00Z' },
    { id: '4', publishDate: '2020-04-15T00:00:00Z' },
  ];

  expect(items.sort(sortByDate)).toMatchSnapshot();
});

test('should fetch channel sections', async () => {
  axios.get.mockResolvedValue({
    data: {
      kind: 'youtube#channelSectionListResponse',
      etag: 'etag-0',
      items: [
        {
          kind: 'youtube#channelSection',
          etag: 'etag-1',
          id: '1',
          snippet: {
            type: 'multiplePlaylists',
            style: 'horizontalRow',
            channelId: 'UCjycPAZuveusvPrk94-ClBw',
            title: 'Popular Series',
            position: 1,
            localized: {
              title: 'Popular Series',
            },
          },
          contentDetails: {
            playlists: [
              'PL8cDVrurCVqMDEP9wiPCgvuZslIGkE_5M',
              'PL8cDVrurCVqN7DvWM3ZI6oHfC5rxBTOJi',
              'PL8cDVrurCVqMZH6QZXigDaZ7HqlStBrvk',
              'PL8cDVrurCVqOAQaI8O8alNwBt2rQp3HYI',
              'PL8cDVrurCVqNrhEI2KZqn2GCyMrq967xV',
              'PL8cDVrurCVqNX0hof1GEmDbIVSHP4xMUs',
            ],
          },
        },
      ],
    },
  });

  const playlist = await fetchChannelSection();

  expect(playlist).toMatchSnapshot();
});

test('should fetch a playlist', async () => {
  axios.get.mockResolvedValue({
    data: {
      items: [
        {
          kind: 'youtube#playlist',
          etag: 'etag',
          id: 'id',
          snippet: {
            channelId: 'UCjycPAZuveusvPrk94-ClBw',
            title: 'Playlist Title',
            description: '',
            thumbnails: [{}],
            channelTitle: 'Echo.Church',
            localized: [{}],
          },
          contentDetails: {
            itemCount: 5,
            videoPublishedAt: '2020-04-04T00:00:00Z',
          },
        },
      ],
    },
  });

  const playlist = await fetchPlaylists([
    'PL8cDVrurCVqMDEP9wiPCgvuZslIGkE_5M',
    'PL8cDVrurCVqN7DvWM3ZI6oHfC5rxBTOJi',
    'PL8cDVrurCVqMZH6QZXigDaZ7HqlStBrvk',
    'PL8cDVrurCVqOAQaI8O8alNwBt2rQp3HYI',
    'PL8cDVrurCVqNrhEI2KZqn2GCyMrq967xV',
    'PL8cDVrurCVqNX0hof1GEmDbIVSHP4xMUs',
  ]);

  expect(playlist).toMatchSnapshot();
});
