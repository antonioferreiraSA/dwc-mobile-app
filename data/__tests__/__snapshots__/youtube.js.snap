// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fetch a playlist 1`] = `
Array [
  Object {
    "id": "id",
    "playlistId": "id",
    "publishDate": "2020-04-04T00:00:00Z",
    "thumbnails": Array [
      Object {},
    ],
    "title": "Playlist Title",
  },
  Object {
    "id": "id",
    "playlistId": "id",
    "publishDate": "2020-04-04T00:00:00Z",
    "thumbnails": Array [
      Object {},
    ],
    "title": "Playlist Title",
  },
  Object {
    "id": "id",
    "playlistId": "id",
    "publishDate": "2020-04-04T00:00:00Z",
    "thumbnails": Array [
      Object {},
    ],
    "title": "Playlist Title",
  },
  Object {
    "id": "id",
    "playlistId": "id",
    "publishDate": "2020-04-04T00:00:00Z",
    "thumbnails": Array [
      Object {},
    ],
    "title": "Playlist Title",
  },
  Object {
    "id": "id",
    "playlistId": "id",
    "publishDate": "2020-04-04T00:00:00Z",
    "thumbnails": Array [
      Object {},
    ],
    "title": "Playlist Title",
  },
  Object {
    "id": "id",
    "playlistId": "id",
    "publishDate": "2020-04-04T00:00:00Z",
    "thumbnails": Array [
      Object {},
    ],
    "title": "Playlist Title",
  },
]
`;

exports[`should fetch channel sections 1`] = `
Array [
  "PL8cDVrurCVqMDEP9wiPCgvuZslIGkE_5M",
  "PL8cDVrurCVqN7DvWM3ZI6oHfC5rxBTOJi",
  "PL8cDVrurCVqMZH6QZXigDaZ7HqlStBrvk",
  "PL8cDVrurCVqOAQaI8O8alNwBt2rQp3HYI",
  "PL8cDVrurCVqNrhEI2KZqn2GCyMrq967xV",
  "PL8cDVrurCVqNX0hof1GEmDbIVSHP4xMUs",
]
`;

exports[`should sort playlist by publish date 1`] = `
Array [
  Object {
    "id": "5",
    "publishDate": "2020-04-20T00:00:00Z",
  },
  Object {
    "id": "4",
    "publishDate": "2020-04-15T00:00:00Z",
  },
  Object {
    "id": "3",
    "publishDate": "2020-04-10T00:00:00Z",
  },
  Object {
    "id": "2",
    "publishDate": "2020-04-05T00:00:00Z",
  },
  Object {
    "id": "1",
    "publishDate": "2020-04-01T00:00:00Z",
  },
]
`;
